### ISSUE TITLE
[Bug] Fix transposed BERT pooler weights in TraditionalBertClassifier

### ISSUE THREAD

**Issue by arun-elr** [vllm-project/semantic-router#2249]
### Describe the bug

In `candle-binding`, `TraditionalBertClassifier` loads the BERT pooler weight (`bert.pooler.dense.weight`) transposed, while the classification head in the same constructor loads its weight without transpose. Only one convention can match HuggingFace. The pooler load is the inconsistent one, so the pooled `[CLS]` representation is computed with a transposed weight matrix and the resulting class probabilities / confidence scores are miscalibrated for the traditional BERT sequence-classification path.

In `candle-binding/src/model_architectures/traditional/bert.rs`, `TraditionalBertClassifier::new`:

```rust
// pooler: transposed
let pooler_weight = vb.get(
    (config.hidden_size, config.hidden_size),
    "bert.pooler.dense.weight",
)?;
let pooler_bias = vb.get(config.hidden_size, "bert.pooler.dense.bias")?;
Linear::new(pooler_weight.t()?, Some(pooler_bias));   // .t()

// classifier head a few lines below: NOT transposed
let classifier_weight = vb.get((num_classes, config.hidden_size), "classifier.weight")?;
let classifier_bias = vb.get(num_classes, "classifier.bias")?;
Linear::new(classifier_weight, Some(classifier_bias)); // no .t()
```

HuggingFace stores `nn.Linear` weights as `[out_features, in_features]`, and `candle_nn::Linear::forward(x)` already computes `x @ weight.t()`, so the correct load is as-is (what the classifier head does). The pooler passes `pooler_weight.t()` (`[in, out]`), so forward effectively computes `x @ W` instead of `x @ W^T`. Because the pooler matrix is square it does not raise a shape error; it silently applies the wrong linear map.

### To Reproduce

1. Load a BERT sequence-classification checkpoint through `TraditionalBertClassifier`.
2. Run a forward pass and read the pooled output / logits / softmax confidence.
3. Compare against HuggingFace `transformers` for the same checkpoint and input.
4. The pooled output and confidences diverge because the pooler weight is applied transposed.

### Expected behavior

Pooled output, logits, and confidences match HuggingFace `transformers` for the same checkpoint and input.

### Affected layer

signal

### Additional context

Pre-existing and unrelated to in-flight feature work; noticed while implementing the cross-encoder reranker (PR #2236 / #2247), which reuses the same BERT + pooler + classifier pattern.

Suggested fix: drop the `.t()` so the pooler matches the classifier head and the HuggingFace convention:

```rust
Linear::new(pooler_weight, Some(pooler_bias));
```

and validate by comparing pooled output / logits against HuggingFace `transformers`. Happy to send a small PR with the one-line fix plus a parity check.

**Comment by arun-elr** [comment]
Maintainers: once this is triaged, please assign it to me. I have full context from the rerank work in #2236 and a concrete fix in mind (drop the pooler transpose and add a HuggingFace parity check), so I can take it forward quickly.

**Comment by Xunzhuo** [comment]
Linked into the H2 signal quality and extractor expansion track: #2335.
